cs244b_project/src/lib.rs

```diff
@@ -435,19 +441,26 @@ impl StreamletInstance {
                                     let mut new_message = message.clone();
                                     // Don't "endorse" this vote unless you've already gotten a proposal 
                                     if vote_this_epoch.is_some() {
+                                        // Make sure hashes match in this VOTE and the stored Proposed block and append signatures if so
                                         if let Some(sig) = self.should_vote(&mut new_message, vote_this_epoch, epoch, &block, &app_interface) {
+                                            if self.seen_block_this_epoch == Some(block.hash) {
+                                                self.sigs_on_seen_block_this_epoch.append(&mut new_message.clone().get_signatures());
+                                                self.sigs_on_seen_block_this_epoch.dedup();
+                                            }
+                                            // update message signature list with the most signatures possible
+                                            new_message.signatures = self.sigs_on_seen_block_this_epoch.clone();
                                             if self.compromise_type != CompromiseType::NoVote {
                                                 // Broadcast messages that we haven't signed yet
```

Comment:
...like, instead of "Broadcast messages we haven't signed yet", it's broadcast messages that have new signatures.